frame/staking/src/lib.rs

```diff
@@ -1769,11 +1923,41 @@ decl_module! {
 			Self::update_ledger(&controller, &ledger);
 		}
 
-		/// Set history_depth value.
+		/// Set `HistoryDepth` value. This function will delete any history information
+		/// when `HistoryDepth` is reduced.
+		///
+		/// Parameters:
+		/// - `new_history_depth`: The new history depth you would like to set.
+		/// - `era_items_deleted`: The number of items that will be deleted by this dispatch.
+		///    This should report all the storage items that will be deleted by clearing old
+		///    era history. Needed to report an accurate weight for the dispatch. Trusted by
+		///    `Root` to report an accurate number.
```

Comment:
items is used below for two different usage:
* as `E` number of history depths removed in `30 * WEIGHT_PER_MICROS).saturating_mul(items)`
* as written above in `.saturating_add(T::DbWeight::get().reads_writes(items, items))`

But actually generally `E` is very less than `items`

Reply:
Yes, good catch. I will remove `.saturating_add((30 * WEIGHT_PER_MICROS).saturating_mul(items))`

as it should be accounted for in the overall weight of 1 read + 1 write